sdk/core/Azure.Core/src/DynamicData/AllowListConverterFactory.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core.Dynamic;
+
+namespace Azure.Core.Json
+{
+    internal class AllowListConverterFactory : JsonConverterFactory
```

Comment:
Do we execute these checks only in our dynamic APIs or also in MutableJsonDocument? I worry about perf impact on MJD.

Reply:
It should be possible to scope it to DynamicData only -- i.e. DynamicData would pass in options with this converter attached, and do type checks prior to calling set on MJD.  If we do that, MJD then would allow serialization of arbitrary types, but maybe we are ok with that?